configure.ac

```
@@ -541,13 +542,24 @@ AS_IF([test x"$with_ze" != x"no"],
 			[],
 			[$with_ze],
 			[],
-			[AC_DEFINE([HAVE_LIBZE], [1],[ZE support])],
+			[have_ze=1],
 			[], [])
        CPPFLAGS="$CPPFLAGS $ze_CPPFLAGS"
        LDFLAGS="$LDFLAGS $ze_LDFLAGS"
        LIBS="$LIBS $ze_LIBS"],
       [])
 
+AS_IF([test "$have_ze" = "1"],
+      [AC_CHECK_HEADER(drm/i915_drm.h, [], [have_ze=0])]
+      [])
+
+AS_IF([test x"$with_ze" != x"no" && test -n "$with_ze" && test "$have_ze" = "0" ],
+	[AC_MSG_ERROR([ZE support requested but ZE runtime not available.])],
+	[])
+
+AS_IF([test "$have_ze" = "1"],
+      [AC_DEFINE([HAVE_LIBZE], [1], [ZE support])])
```

```suggestion
AC_DEFINE_UNQUOTED([HAVE_LIBZE], [$have_ze], [ZE support])])
```

Then you can always:

```c
#if HAVE_LIBZE
...blah...
#endif
```

Which is always safe (because `HAVE_LIBZE` will always be 0 or 1), vs.:

```c
#ifdef HAVE_L1BZE
...blah...
#endif
```

which will never be true (did you notice the typo?).  ...but the compiler/preprocessor will never warn you about the typo -- it'll just silently always be false.

If you make the typo in the first form (`#if HAVE_L1BZE`), you'll get a preprocessor warning/error.

---

The autotools lord has spoken his autotools wisdom and now I will make it so 🙇 All hail the autotools lord who makes things possible for us plebeians!

---

I always thought he was more of a czar.  But if we're plebeians, maybe he's more of an emperor?

---

All this means is that I have more knowledge than most others in a topic that no one else cares about.  😜